### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by brianfpeters**
Fix Collection.to_dict() to support array of links with same link relation as per section 4.1.1 of draft RFC [https://tools.ietf.org/html/draft-kelly-json-hal-07].

If a link is the second to use a previous relation then the first link's body is moved inside a list and the second link's body is appended to that list.

Otherwise use the dict.upate() function as before.

### CODE REVIEW

**Comment by krak3n** [comment]
Thanks for the PR @brianfpeters 

Looks like the test suite is failing, could you take a look at the failing tests? https://circleci.com/gh/thisissoon/Flask-HAL/68

You can run the test suite locally, the `circle.yml` has the basic instructions, we use `docker` to run it.

**Comment by krak3n** [comment]
@thisissoon/members could you guys take a look at this one please.

**Comment by radeklos** [comment]
can you also add some tests?

**Comment by radeklos** [comment]
@krak3n @brianfpeters  tests are failing because `ERROR: py26: InterpreterNotFound: python2.6`.. `ikalnitsky/pythonista` dropped a support for py26.

**Comment by radeklos** [comment]
@brianfpeters we just dropped support for `py26` can you please rebase your `develop`?

**Comment by brianfpeters** [comment]
I have rebased, but as I am new to github you may want to review carefully. Is this pull request (#20) now ok or do I have to make a new pull request?

**Comment by radeklos** [comment]
@brianfpeters  The rebased fixed the `py26` issue. When you click to the red cross it navigates you to circle which says you what is wrong now.. You have some `IndentationError` errors.

File: flask_hal/link.py [hunk 1]
```diff
@@ -86,7 +86,12 @@ def to_dict(self):
         links = {}
 
         for link in self:
-            links.update(link.to_dict())
+            if link.rel in links.keys():
+                if isinstance(links[link.rel], dict):
+                    links[link.rel] = [ links[link.rel] ]
+               links[link.rel].append(link.to_dict()[link.rel])
```
Comment (radeklos) [inline flask_hal/link.py:92]: One space is missing from left.

**Comment by krak3n** [comment]
Seems this has moved to a new PR #22

Closing.

**Pull request closed without merging**